File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    count = 0;
    while count < 500 and num != 1:
        if num % 2 == 0:
            num /= 2
        else:
            num = num*3+1
        count += 1
    if count == 500:
        return -1
    else:
        return count

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        if (num == 1) {
            return 0;
        }
        long n = num;
        int i = 0;
        while (i < 500 && n != 1) {
            if (n % 2 == 0) {
                n /= 2;
            } else {
                n = n * 3 + 1;
            }
            i++;
        }
        answer = n == 1 ? i : -1;
        return answer;
    }
}